### PR TITLE
create ObjectHolder

### PR DESCRIPTION
**Pull request by vasiliypekhterev**

### CODE REVIEW

File: src/main/java/ua/artcode/udiary/dao/RecordDaoJsonImpl.java [hunk 1]
```diff
@@ -11,9 +12,13 @@
 
 public class RecordDaoJsonImpl implements RecordDao {
 
-    private Gson gson = new GsonBuilder().setPrettyPrinting().create();
+    private Gson gson;
     private static final String RECORDS_PATH = "./src/ua/artcode/udiary/resources/record/records.txt";
```
Comment (presly808) [inline src/main/java/ua/artcode/udiary/dao/RecordDaoJsonImpl.java:16]: get from the properties

File: src/main/java/ua/artcode/udiary/config/ObjectHolder.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import java.util.List;
+import java.util.Map;
+
+public class ObjectHolder {
+
+    private static Map<String,Object> staticObjects = new HashMap<>();
+
+    static  {
+        staticObjects.put("gson", new GsonBuilder().setPrettyPrinting().create());
+        staticObjects.put("userListType",new TypeToken<List<User>>(){}.getType());
```
Comment (presly808) [inline src/main/java/ua/artcode/udiary/config/ObjectHolder.java:18]: perhaps we can avoid userListType in map

File: src/main/java/ua/artcode/udiary/dao/UserDaoJsonImpl.java [hunk 3]
```diff
@@ -18,16 +19,16 @@
     // todo get path from config
     private final String DATA_PATH;
 
-    // todo GSON must be a singleton
-    private Gson gson = new GsonBuilder().setPrettyPrinting().create();
-
-    // todo init in constructor
-    private Type userListType = new TypeToken<List<User>>(){}.getType();
+    Gson gson;
```
Comment (presly808) [inline src/main/java/ua/artcode/udiary/dao/UserDaoJsonImpl.java]: fields should be private. it doesn't have sense to make fiels package private in our case

**Review by presly808 — COMMENTED** [review]
Also try to change all places where we may use objectHolder

File: src/main/java/ua/artcode/udiary/dao/UserDaoJsonImpl.java (hunk 3 above)
Comment (vasiliypekhterev) [reply src/main/java/ua/artcode/udiary/dao/UserDaoJsonImpl.java]: Fixed.

File: src/main/java/ua/artcode/udiary/dao/RecordDaoJsonImpl.java (hunk 1 above)
Comment (vasiliypekhterev) [reply src/main/java/ua/artcode/udiary/dao/RecordDaoJsonImpl.java:16]: RecordDao json implementation will be changed. I will fix it after that.

File: src/main/java/ua/artcode/udiary/config/ObjectHolder.java (hunk 2 above)
Comment (vasiliypekhterev) [reply src/main/java/ua/artcode/udiary/config/ObjectHolder.java:18]: Unfortunately, I dont understatnd the point of this comment. More explanation is needed.

**Comment by vasiliypekhterev** [comment]
Moved ConfigHolder to the ObjectHolder. I think other changes could be done on separate cards by their owners to avoid merge conflicts.

**Pull request merged**